pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapper.java

```diff
@@ -85,14 +89,45 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
         "Wrong number of arguments for method: %s, expected: %s, actual: %s", _functionInvoker.getMethod(),
         parameterTypes.length, numArguments);
 
-    _arguments = new Object[numArguments];
+    _scalarArguments = new Object[numArguments];
     _nonLiteralIndices = new int[numArguments];
     _nonLiteralFunctions = new TransformFunction[numArguments];
     for (int i = 0; i < numArguments; i++) {
       TransformFunction transformFunction = arguments.get(i);
       if (transformFunction instanceof LiteralTransformFunction) {
-        String literal = ((LiteralTransformFunction) transformFunction).getStringLiteral();
-        _arguments[i] = parameterTypes[i].convert(literal, PinotDataType.STRING);
+        LiteralTransformFunction literalTransformFunction = (LiteralTransformFunction) transformFunction;
+        DataType dataType = literalTransformFunction.getResultMetadata().getDataType();
```

Comment:
Not introduced in this PR, but can literal be `INT` or `FLOAT` type? `NumberUtils.createNumber(literal)` will parse to the string to the smallest type, and we should probably keep this convention

Reply:
Yes +1